### PR TITLE
Updated to akka 2.6.5, akka http to 10.1.11, akka-stream-contrib to 0…

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

Upgrade to Akka 2.6.5.
runLocal for taxi-ride app does not work yet since Flink depends on Akka 2.5.

### Why are the changes needed?

Support latest. This will also make it possible to add external sharding strategy.

### Does this PR introduce any user-facing change?

Yes, materializer is not necessary anymore, can use system in most places.

### How was this patch tested?

- [x] unit tests and runLocal.
- [x] Removed use of cloudflow-events in runner for test for now, which uses Akka (through use of Skuber).
- [x] Tested on GKE, taxi-ride and CRA app.

### CODE REVIEW

File: core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/javadsl/OutletTap.scala [hunk 1]
```diff
@@ -48,7 +50,7 @@ case class ProbeOutletTap[T](outlet: CodecOutlet[T])(implicit system: ActorSyste
       .alsoTo(
         Flow[PartitionedValue[T]]
           .map(pv ⇒ Pair(pv.key, pv.value))
-          .to(Sink.actorRef[Pair[String, T]](probe.getTestActor, Completed))
+          .to(Sink.actorRef[Pair[String, T]](probe.getTestActor, Completed, Failed))
```
Comment (olofwalker) [inline core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/javadsl/OutletTap.scala:53]: Completed -> Failed ?

File: core/cloudflow-akka-util/src/test/scala/cloudflow/akkastream/util/scaladsl/SplitterSpec.scala [hunk 2]
```diff
@@ -49,13 +47,13 @@ class SplitterSpec extends WordSpec with MustMatchers with ScalaFutures with Bef
       }
 
       def runnableGraph =
-        sourceWithOffsetContext(in).to(Splitter.sink(flow, left, right))
+        sourceWithCommittableContext(in).to(Splitter.sink(flow, left, right))
```
Comment (olofwalker) [inline core/cloudflow-akka-util/src/test/scala/cloudflow/akkastream/util/scaladsl/SplitterSpec.scala:50]: Did this changed name or something?

Comment (RayRoestenburg) [reply core/cloudflow-akka-util/src/test/scala/cloudflow/akkastream/util/scaladsl/SplitterSpec.scala:50]: yes

File: core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/javadsl/OutletTap.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/javadsl/OutletTap.scala:53]: it's new API you have to provide what the failure message will be too.

Comment (olofwalker) [reply core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/javadsl/OutletTap.scala:53]: Ignore it, in the split lane it looked like you replaced Completed with Failed.

File: core/project/Dependencies.scala [hunk 3]
```diff
@@ -22,6 +22,8 @@ object Library {
   val AkkaStreamContrib     = "com.typesafe.akka"     %% "akka-stream-contrib"       % "0.10"
   val AkkaStreamKafka       = "com.typesafe.akka"     %% "akka-stream-kafka"         % Version.AlpakkaKafka exclude("com.fasterxml.jackson.core","jackson-databind") exclude("com.fasterxml.jackson.module", "jackson-module-scala")
   val AkkaStreamKafkaTestkit = "com.typesafe.akka"    %% "akka-stream-kafka-testkit" % Version.AlpakkaKafka exclude("com.typesafe.akka", "akka-stream-testkit")
+  val AkkaStreamTestkit      = "com.typesafe.akka"   %% "akka-stream-testkit"        % Version.Akka         
+  
   val AkkaCluster           = "com.typesafe.akka"     %% "akka-cluster"              % Version.Akka
```
Comment (patriknw) [inline core/project/Dependencies.scala:27]: Artery TCP is now default, so the https://github.com/lightbend/cloudflow/blob/master/core/cloudflow-akka/src/main/resources/akka-cluster-k8.conf#L32 should be updated to use Artery config. See https://doc.akka.io/docs/akka/current/remoting-artery.html#configuration

Comment (patriknw) [reply core/project/Dependencies.scala:27]: also, if you have any `akka.tcp` addresses they should be replaced with `akka`. see details in migration guide https://doc.akka.io/docs/akka/current/project/migration-guide-2.5.x-2.6.x.html#migrating-from-classic-remoting-to-artery

Comment (RayRoestenburg) [reply core/project/Dependencies.scala:27]: Thanks Patrik!

File: core/project/Dependencies.scala [hunk 4]
```diff
@@ -33,34 +35,33 @@ object Library {
   val Config                = "com.typesafe"           % "config"                   % "1.3.4"
   val Logback               = "ch.qos.logback"         % "logback-classic"          % "1.2.3"
 
-  val SprayJson             = "io.spray"              %% "spray-json"               % "1.3.5"
-  val Bijection             = "com.twitter"           %% "bijection-avro"           % "0.9.6"
+  val SprayJson              = "io.spray"              %% "spray-json"               % "1.3.5"
+  val Bijection              = "com.twitter"           %% "bijection-avro"           % "0.9.6"
 
   val JacksonScalaModule    = "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.10.0"
   val JacksonDatabind       = "com.fasterxml.jackson.core"    % "jackson-databind"     % "2.9.9.1"
```
[the remark this answers is not in the document]
Comment (RayRoestenburg) [reply core/project/Dependencies.scala]: Fixed.

File: core/project/Dependencies.scala (hunk 3 above)
Comment (RayRoestenburg) [reply core/project/Dependencies.scala:27]: Fixed.

**Pull request merged**